logic/src/commonMain/kotlin/com/wire/kalium/logic/data/publicuser/PublicUserRepository.kt

```diff
@@ -0,0 +1,57 @@
+package com.wire.kalium.logic.data.publicuser
+
+import com.wire.kalium.logic.CoreFailure
+import com.wire.kalium.logic.NetworkFailure
+import com.wire.kalium.logic.data.publicuser.model.PublicUserSearchResult
+import com.wire.kalium.logic.functional.Either
+import com.wire.kalium.logic.functional.mapLeft
+import com.wire.kalium.logic.functional.suspending
+import com.wire.kalium.logic.wrapApiRequest
+import com.wire.kalium.network.api.contact.search.ContactSearchApi
+import com.wire.kalium.network.api.contact.search.ContactSearchRequest
+import com.wire.kalium.network.api.user.details.ListUserRequest
+import com.wire.kalium.network.api.user.details.UserDetailsApi
+import com.wire.kalium.network.api.user.details.qualifiedIds
+
+
+interface PublicUserRepository {
+    suspend fun searchPublicContact(
+        searchQuery: String,
+        domain: String,
+        maxResultSize: Int? = null
+    ): Either<CoreFailure, PublicUserSearchResult>
+}
+
+class PublicUserRepositoryImpl(
+    private val contactSearchApi: ContactSearchApi,
+    private val userApi: UserDetailsApi,
+    private val publicUserMapper: PublicUserMapper,
+) : PublicUserRepository {
+
+    override suspend fun searchPublicContact(
+        searchQuery: String,
+        domain: String,
+        maxResultSize: Int?
+    ): Either<NetworkFailure, PublicUserSearchResult> {
+        return suspending {
+            wrapApiRequest {
+                contactSearchApi.search(
+                    ContactSearchRequest(
+                        searchQuery = searchQuery,
+                        domain = domain,
+                        maxResultSize = maxResultSize
+                    )
+                )
+            }.flatMap { contactResultValue ->
+                wrapApiRequest {
+                    userApi.getMultipleUsers(ListUserRequest.qualifiedIds(contactResultValue.documents.map { it.qualifiedID }))
+                }.map { userDetailsResponses ->
+                    PublicUserSearchResult(
+                        totalFound = contactResultValue.found,
+                        publicUsers = publicUserMapper.fromUserDetailResponses(userDetailsResponses)
+                    )
+                }
+            }
+        }.mapLeft { NetworkFailure.ServerMiscommunication(it.kaliumException) }
```

Comment:
```suggestion
```
No need to `mapLeft` here since `flatMap` +  `wrapApiRequest` will catch the errors from the new request anyway

Reply:
true ! thanks